api/resmap/resmap.go

```diff
@@ -30,7 +30,7 @@ type TransformerWithProperties struct {
 
 // A Generator creates an instance of ResMap.
 type Generator interface {
-	Generate() (ResMap, error)
+	Generate() (ResMap, []string, error)
```

Comment:
So unfortunately, this is a public interface and we can't change the signature (this is how users built legacy style extensions and we can't break that just yet). 

I think it also isn't quite sensical for this Generator to return an array of keys just for a ConfigMap. That type of data is specific to the ConfigMap generator and is not generalizable to all generators, so should instead be attached to the ConfigMapGenerator and methods - and only to things that are specific to the ConfigMapGenerator.

Reply:
I would think about other solution to see if we change limit the changes only to configmapgenerator